src/store/reducers/deploymentsReducer.js

```diff
@@ -89,24 +121,57 @@ const deleteDeploymentReducer = (state, { payload, fromAction }) => {
 const updateDeploymentsReducer = (state, { payload, fromAction }) => {
   const { entities: { deployments }, result } = normalize(payload, deploymentListSchema);
   return update(state, {
-    entities: { $set: deployments },
+    entities: { deployments: { $set: deployments } },
     items: { $set: result },
     lastUpdated: { $set: moment() },
     ...setPending(fromAction.type, false)
   });
 };
 
+const updateDeploymentReducer = (state, { payload, fromAction }) => {
+  const { deviceStatuses } = payload || {};
+  return update(state, {
+    currentDeployment: { $set: payload },
+    deviceStatuses: { $set: deviceStatuses },
+    currentDeploymentLastUpdated: { $set: moment() },
+    ...setPending(fromAction.type, false)
+  });
+};
+
+const updateDeployedDevicesReducer = (state, { payload: [modules, devices], fromAction }) => {
+  const normalizedDevices = normalize(devices, deployedDevicesListSchema).entities.deployedDevices;
+  const normalizedModules = normalize(modules, deployedDevicesListSchema).entities.deployedDevices;
+  const deployedDevices = Object.keys(normalizedDevices)
+    .reduce(
```

Comment:
This is more what I had in mind. A clear distinction between when the function opens, closes and there the parameters are.
```javascript
.reduce(
  (acc, deviceId) => ({
    ...acc,
    [deviceId]: { 
      ...(acc[deviceId] || {}), 
      firmware: normalizedDevices[deviceId].firmware 
    }
  }),
  normalizedModules
);
```